components/formats-gpl/src/loci/formats/in/ZeissCZIReader.java

```diff
@@ -425,14 +425,24 @@ else if (rawData.length == (realX + 1) * (realY + 1) * pixel) {
             }
           }
           else {
-            byte[] rawData = new SubBlock(plane).readPixelData();
-            RandomAccessInputStream s = new RandomAccessInputStream(rawData);
-            try {
-              readPlane(s, x, y, w, h, realX - getSizeX(), buf);
-              emptyTile = false;
+            byte[] rawData = null;
+            // re-use the existing stream if we know there is only one file
+            // this saves a little time over opening a new stream for every tile/plane
+            if (pixels.size() == 0) {
+              rawData = new SubBlock(plane).readPixelData(in, new Region(x, y, w, h), buf);
+            }
+            else {
+              rawData = new SubBlock(plane).readPixelData();
             }
-            finally {
-              s.close();
+            if (rawData.length > buf.length) {
+              RandomAccessInputStream s = new RandomAccessInputStream(rawData);
```

Comment:
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html could be used here to avoid the ``try/finally``